spec/models/equipment_item_spec.rb

```diff
@@ -1,160 +1,269 @@
     # delete the call?
-    it 'saves an empty string value as nil for deleted_at field' do
-      @item.deleted_at = '   '
-      @item.save
-      expect(@item.deleted_at).to eq(nil)
+    item = FactoryGirl.build(:equipment_item)
+    item.deleted_at = '   '
+    item.save
+    expect(item.deleted_at).to eq(nil)
+  end
+
+  describe '#active' do
+    it 'returns active equipment items' do
+      active = FactoryGirl.create(:equipment_item)
+      FactoryGirl.create(:deactivated)
+      expect(described_class.active).to match_array([active])
+    end
+  end
+
+  describe '#for_eq_model' do
+    it 'counts the number of items for the given model' do
+      items = Array.new(2) { |i| EquipmentItemMock.new(equipment_model_id: i) }
+      expect(described_class.for_eq_model(0, items)).to eq(1)
     end
   end
 
-  describe '.active' do
-    before(:each) do
-      @active = FactoryGirl.create(:equipment_item)
-      @deactivated = FactoryGirl.create(:deactivated)
+  describe '#status' do
+    it "returns 'Deactivated' when deleted_at is set" do
+      item = FactoryGirl.build_stubbed(:equipment_item,
+                                       deleted_at: Time.zone.today)
+      expect(item.status).to eq('Deactivated')
+    end
+    it "returns 'available' when active and not currently checked out" do
+      item = FactoryGirl.build_stubbed(:equipment_item)
+      expect(item.status).to eq('available')
     end
+    it 'includes reservation information when checked out' do
+      res = FactoryGirl.create(:checked_out_reservation)
+      item = res.equipment_item
+      expect(item.status).to include('checked out by')
+    end
+    it 'includes deactivation reason if it is set' do
+      reason = 'because i can'
+      item = FactoryGirl.build_stubbed(:equipment_item,
+                                       deleted_at: Time.zone.today,
+                                       deactivation_reason: reason)
+      expect(item.status).to include(reason)
+    end
+  end
 
-    it 'Should return all active equipment items' do
-      expect(EquipmentItem.active).to include(@active)
+  describe '#current_reservation' do
+    it 'returns nil if no associated reservation' do
+      item = FactoryGirl.build_stubbed(:equipment_item)
+      expect(item.current_reservation).to be_nil
     end
+    it 'returns the reservation that currently has the item checked out' do
+      res = FactoryGirl.create(:checked_out_reservation)
+      item = res.equipment_item
+      expect(item.current_reservation).to eq(res)
+    end
+  end
 
-    it 'Should not return inactive equipment items' do
-      expect(EquipmentItem.active).not_to include(@deactivated)
+  describe '#available?' do
+    it 'returns true when the status is available' do
+      item = FactoryGirl.create(:equipment_item)
+      expect(item.available?).to be_truthy
+    end
+    it 'returns false if when the status is not available' do
+      res = FactoryGirl.create(:checked_out_reservation)
```

Comment:
Same here, not sure we need an actually persisted reservation (but again, it's fine if it would be messy to stub).